services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/CopyObjectHelper.java

```diff
@@ -128,6 +126,10 @@ private void doCopyInParts(CopyObjectRequest copyObjectRequest,
                                String uploadId) {
 
         long optimalPartSize = genericMultipartHelper.calculateOptimalPartSizeFor(contentLength, partSizeInBytes);
+        if (optimalPartSize > partSizeInBytes) {
+            log.info(() -> String.format("Configured partSize is %d, but using %d to prevent reaching maximum number of parts "
+                                         + "allowed", partSizeInBytes, optimalPartSize));
```

Comment:
Info is a pretty loud log level. Is this necessary?

Reply:
This is only happens when `total size / part size > 10,000` which will result in not using the configured part size set by the user (to not hit the 10,000 part limit imposed by s3), I thought being a little loud about it was a good idea.

Reply:
Should we have a flag that the user opts-in to allow this to happen, so that we don't have to warn when it does? (They've consented to it, after all!)

Logging above debug for anything that isn't an error scares me, since we don't know what the user's throughput is like, and we frequently get people upset about us filling their logs.

Reply:
I see, I can lower to debug then